tests/MyGame/Example/ArrayStructT.java

```diff
@@ -0,0 +1,52 @@
+// automatically generated by the FlatBuffers compiler, do not modify
+
+package MyGame.Example;
+
+import java.nio.*;
+import java.lang.*;
+import java.util.*;
+import com.google.flatbuffers.*;
+
+public class ArrayStructT {
+  private float a;
+  private int[] b;
+  private byte c;
+  private MyGame.Example.NestedStructT[] d;
+  private int e;
+  private long[] f;
+
+  public float getA() { return a; }
+
+  public void setA(float a) { this.a = a; }
+
+  public int[] getB() { return b; }
+
+  public void setB(int[] b) { this.b = b; }
+
+  public byte getC() { return c; }
+
+  public void setC(byte c) { this.c = c; }
+
+  public MyGame.Example.NestedStructT[] getD() { return d; }
+
+  public void setD(MyGame.Example.NestedStructT[] d) { this.d = d; }
+
+  public int getE() { return e; }
+
+  public void setE(int e) { this.e = e; }
+
+  public long[] getF() { return f; }
+
+  public void setF(long[] f) { this.f = f; }
+
+
+  public ArrayStructT() {
+    this.a = 0.0f;
+    this.b = new int[15];
```

Comment:
this allocation seems to get overwritten by `unPackTo`.. either it should be set to `null`, or the unpack code should just write to the existing array. Probably the latter, since we want to allow users to create legit `ArrayStructT` instances themselves too.

Reply:
Actually, you raised another very delicate point.
The current setter doesn't even check the length of the input array, which means that the Struct can receive larger/smaller arrays than the accepted.

As I can see it there are 3 options:
1. Add length validations for structs with fixed size arrays, init the array at the constructor to null, and set the pointer to the input array.
2. Add length validations for structs with fixed size arrays and make shallow copy of the input array into the received array.
3. Change the setter signature to accept a a single element and it's required position in the array.

From my experience in Java, option 1 is the most common.

Reply:
1 sounds best.

Reply:
@aardappel thinking about this a bit more I found even better solution.
4. Add length validations for structs with fixed size arrays, init the array at the constructor to the array of the specified size, and set the pointer to the input array. In unpackTo I would do as you suggested and write to existing array, and leverage the fact that Java returns array pointers by value, thus instead of generating a new array will just get the current pointer and modify that array.

The fact that the struct is initialized with an array conserves the invariant that arrays always exist in FlatBuffers even if they do not initialized by hand. Making that pointer null seems incorrect for object which FlatBuffers defines as always existing.

Reply:
sounds good, but will have to see what that looks like.

Reply:
@aardappel Implemented it. I added a comment the changed code so you can see how it looks.